fees/betswirl/index.ts

```diff
@@ -0,0 +1,212 @@
+                    partnerAmount
+                    treasuryAmount
+                    teamAmount
+                }
+                yesterdayTokens: tokens(block: { number: ${yesterdayLastBlock} }) {
+                    id
+                    dividendAmount
+                    bankAmount
+                    partnerAmount
+                    treasuryAmount
+                    teamAmount
+                }
+            }`
+      );
+
+      const dailyUserFees: any = {};
+      const dailyFees: any = {};
+      const dailyRevenue: any = {};
+      const dailyProtocolRevenue: any = {};
+      const dailyHoldersRevenue: any = {};
+      const dailySupplySideRevenue: any = {};
+      const totalUserFees: any = {};
+      const totalFees: any = {};
+      const totalRevenue: any = {};
+      const totalProtocolRevenue: any = {};
+      const totalDailyHoldersRevenue: any = {};
+      const totalSupplySideRevenue: any = {};
+      for (const token of graphRes.tokens) {
+        const tokenKey = chain + `:` + token.id;
+
+        totalUserFees[tokenKey] = fromWei(
+          toBN(token.dividendAmount)
+            .plus(token.bankAmount)
+            .plus(token.partnerAmount)
+            .plus(token.treasuryAmount)
+            .plus(token.teamAmount)
+        ).toNumber();
+        totalFees[tokenKey] = totalUserFees[tokenKey];
+
+        totalSupplySideRevenue[tokenKey] = fromWei(
+          toBN(token.bankAmount).plus(token.partnerAmount)
+        ).toNumber();
+
+        totalProtocolRevenue[tokenKey] = fromWei(
+          toBN(token.treasuryAmount).plus(token.teamAmount)
+        ).toNumber();
+        totalDailyHoldersRevenue[tokenKey] = fromWei(
+          token.dividendAmount
+        ).toNumber();
+        totalRevenue[tokenKey] =
+          totalProtocolRevenue[tokenKey] + totalDailyHoldersRevenue[tokenKey];
+      }
+
+      for (const token of graphRes.yesterdayTokens) {
+        const tokenKey = chain + `:` + token.id;
+
+        dailyUserFees[tokenKey] =
+          totalUserFees[tokenKey] -
+          fromWei(
+            toBN(token.dividendAmount)
+              .plus(token.bankAmount)
+              .plus(token.partnerAmount)
+              .plus(token.treasuryAmount)
+              .plus(token.teamAmount)
+          ).toNumber();
+        dailyFees[tokenKey] = dailyUserFees[tokenKey];
+
+        dailyHoldersRevenue[tokenKey] =
+          totalDailyHoldersRevenue[tokenKey] -
+          fromWei(token.dividendAmount).toNumber();
+        dailyProtocolRevenue[tokenKey] =
+          totalProtocolRevenue[tokenKey] -
+          fromWei(toBN(token.treasuryAmount).plus(token.teamAmount)).toNumber();
+        dailyRevenue[tokenKey] =
+          dailyHoldersRevenue[tokenKey] + dailyProtocolRevenue[tokenKey];
+
+        dailySupplySideRevenue[tokenKey] =
+          totalSupplySideRevenue[tokenKey] -
+          fromWei(toBN(token.bankAmount).plus(token.partnerAmount)).toNumber();
+      }
+
+      return {
```

Comment:
all value should sum to usd amoumt, ex. dailyFees: '10' // mean 10 usd

Reply:
Do you know how I can retrieve the price from CG or CMC or another API?